### PR TITLE
MAINT: removing the deprecated submodule

### PR DESCRIPTION
**Pull request by bsipocz**
The functionality has been made private a while ago, and it's probably time to remove the submodule.

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -5243,8 +5243,6 @@
 
     Examples
     --------
-    >>> np.core.umath_tests.matrix_multiply.signature
-    '(m,n),(n,p)->(m,p)'
     >>> np.linalg._umath_linalg.det.signature
     '(m,m)->()'
     >>> np.add.signature is None
```
Comment (seberg) [inline numpy/core/_add_newdocs.py:5248]:
```suggestion
    >>> np.matmul.signature
    '(n?,k),(k,m?)->(n?,m?)'
    >>> np.add.signature is None
```
A bit more complicated with the ?, but maybe that is fine...

**Review by seberg — APPROVED** [review]
Thanks, LGTM either way.  I had misunderstood that we even had a python stub without the underscore...

**Comment by bsipocz** [comment]
> I had misunderstood that we even had a python stub without the underscore...

The beauty of collecting all docstrings for testing 🙂

**Comment by seberg** [comment]
Oh, forgot, could add a bullet point into `expired` category.  But I don't care, this is niche, since the "test" in the name should give idea...

**Pull request merged**